internal/backend/svc/svc.go

```diff
@@ -207,20 +208,25 @@ func makeFxOpts(cfg *Config, opts RunOptions) []fx.Option {
 				// Replace the original mux with a main mux and also expose the original mux as the no-auth one.
 				authed := http.NewServeMux()
 
-				wrap := func(h http.Handler) http.Handler {
-					// TODO(ENG-3): Wrap the handler with auth middleware.
-					return h
-				}
-
 				mux = svc.Mux()
 
-				mux.Handle("/", wrap(authed))
+				mux.Handle("/", authenticator.Middleware(authed))
```

Comment:
Question: if Descope is enabled, but I haven't logged in yet, or my login failed, won't this prevent me from seeing anything? Won't it affect images, CSS and scripts? Won't it affect integration webhooks?

Reply:
I might have put the authenticated route on the wrong place,
technically we would only want to block APIs and not web pages / standard http which is not grpc
I'll have a look

Reply:
fixed it so only routes that begins with /api/ would be authenticated
next PR once this is approved is updating all our grpc endpoints to move to the authenticated routes